vyos/vyos-1x
# OpenVPN CLI-option: T6571: rename ncp-ciphers with data-ciphers

## Change Summary

Rename ncp-ciphers with data-ciphers. This option was called --ncp-ciphers in OpenVPN 2.4 but has been renamed to --data-ciphers in OpenVPN 2.5 to more accurately reflect its meaning.
## Types of changes


## Related Task(s)


https://vyos.dev/T6571
## Related PR(s)


## Component(s) name

OpenVPN
## Proposed changes

Previous format:

```
vyos@testing# set int openvpn vtun10 encryption ncp-ciphers
Possible completions:
   none                 Disable encryption
   3des                 DES algorithm with triple encryption
   aes128               AES algorithm with 128-bit key CBC
   aes128gcm            AES algorithm with 128-bit key GCM
```

New name "data-ciphers":

```
vyos@testing# set int openvpn vtun10 encryption data-ciphers
Possible completions:
   none                 Disable encryption
   3des                 DES algorithm with triple encryption
   aes128               AES algorithm with 128-bit key CBC
   aes128gcm            AES algorithm with 128-bit key GCM
```
## How to test

Run tab to see the completion help with this command "set int openvpn vtun10 encryption"

```
set int openvpn vtun1 encryption
Possible completions:
   cipher               Standard Data Encryption Algorithm
+  data-ciphers          Cipher negotiation list for use in server or client mode
```

## Smoketest result

```
vyos@testing:/usr/libexec/vyos/tests/smoke/cli$ sudo systemctl stop vyos-configd
vyos@testing:/usr/libexec/vyos/tests/smoke/cli$ ./test_interfaces_openvpn.py    test_openvpn_client_interfaces (__main__.TestInterfacesOpenVPN.test_openvpn_client_interfaces) ... ERROR
test_openvpn_client_verify (__main__.TestInterfacesOpenVPN.test_openvpn_client_verify) ... ok
test_openvpn_options (__main__.TestInterfacesOpenVPN.test_openvpn_options) ... ok
test_openvpn_server_subnet_topology (__main__.TestInterfacesOpenVPN.test_openvpn_server_subnet_topology) ... ok
test_openvpn_server_verify (__main__.TestInterfacesOpenVPN.test_openvpn_server_verify) ... ok
test_openvpn_site2site_interfaces_tun (__main__.TestInterfacesOpenVPN.test_openvpn_site2site_interfaces_tun) ... ok
test_openvpn_site2site_verify (__main__.TestInterfacesOpenVPN.test_openvpn_site2site_verify) ... ok

======================================================================
ERROR: test_openvpn_client_interfaces (__main__.TestInterfacesOpenVPN.test_openvpn_client_interfaces)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/libexec/vyos/tests/smoke/cli/./test_interfaces_openvpn.py", line 213, in test_openvpn_client_interfaces
    self.cli_commit()
  File "/usr/libexec/vyos/tests/smoke/cli/base_vyostest_shim.py", line 83, in cli_commit
    self._session.commit()
  File "/usr/lib/python3/dist-packages/vyos/configsession.py", line 210, in commit
    out = self.__run_command([COMMIT])
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/vyos/configsession.py", line 147, in __run_command
    raise ConfigSessionError(output)
vyos.configsession.ConfigSessionError: [ pki ]
ConfigError('CA certificates are not a valid chain')

[[pki]] failed
[ interfaces openvpn vtun10 ]
CA certificates are not a valid chain

[[interfaces openvpn vtun10]] failed
Commit failed


----------------------------------------------------------------------
Ran 7 tests in 235.023s

FAILED (errors=1)
```
## Checklist:
